worker.go

```diff
@@ -288,12 +243,15 @@ func go_frankenphp_worker_handle_request_start(threadIndex C.uintptr_t) C.bool {
 		if c := logger.Check(zapcore.DebugLevel, "shutting down"); c != nil {
 			c.Write(zap.String("worker", thread.worker.fileName))
 		}
-		thread.worker = nil
-		executePHPFunction("opcache_reset")
+
+		// execute opcache_reset if the restart was triggered by the watcher
+		if watcherIsEnabled && workersAreRestarting.Load() && !executePHPFunction("opcache_reset") {
```

Comment:
Hmmm... what if my worker script takes too long to start up? `workersAreRestarting` could be `false` by that point.

Reply:
Sorry, I accidentally pushed my changes to the branch and quickly deleted the commit via force push. But yes, this epically crashes when opcache is not installed:

```
Stack trace:
#0 /home/withinboredom/code/frankenphp/testdata/_executor.php(9): frankenphp_handle_request()
#1 /home/withinboredom/code/frankenphp/testdata/index.php(3): require_once('...')
#2 {main}
  thrown in /home/withinboredom/code/frankenphp/testdata/_executor.php on line 9        {"syslog_level": "err"}
2024/11/16 10:25:07.003 ERROR   PHP Fatal error:  Uncaught Error: Invalid callback opcache_reset, function "opcache_reset" not found or invalid function name in /home/withinboredom/code/frankenphp/testdata/_executor.php:9
```

Reply:
Also, this happens on `main`.

Reply:
Good job on finding the missing function error 👍 , I didn't know that one couldn't be caught.

`opcache_reset` will be executed in the middle of the worker script here (not on startup), right before `frankenphp_handle_request()` returns `false`. I did a lot of testing and this works pretty well. 
I don't know if executing it a the end of the script would be better, but I also didn't know how to best 'hook into' the end of the worker script.